### PR TITLE
feat: --network-dataplane flag

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
**Description**
This pr gates the node labels we add to be gated alongside network dataplane only adding the cilium labels when cilium is enabled. 
**How was this change tested?**
- make presubmit

**Release Note**

```release-note
introducing the --network-dataplane flag
```

### CODE REVIEW

File: pkg/operator/options/options.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ func (o *Options) AddFlags(fs *coreoptions.FlagSet) {
 	fs.StringVar(&o.SSHPublicKey, "ssh-public-key", env.WithDefaultString("SSH_PUBLIC_KEY", ""), "[REQUIRED] VM SSH public key.")
 	fs.StringVar(&o.NetworkPlugin, "network-plugin", env.WithDefaultString("NETWORK_PLUGIN", "azure"), "The network plugin used by the cluster.")
 	fs.StringVar(&o.NetworkPolicy, "network-policy", env.WithDefaultString("NETWORK_POLICY", ""), "The network policy used by the cluster.")
+	fs.StringVar(&o.NetworkDataplane, "network-dataplane", env.WithDefaultString("NETWORK_DATAPLANE", "cilium"), "The network dataplane used by the cluster.")
```
Comment (Bryce-Soghigian) [inline pkg/operator/options/options.go:84]: defaulting to cilium may be the wrong way to do things? We could default to empty.

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

File: pkg/operator/options/options_validation.go [hunk 2]
```diff
@@ -44,6 +45,12 @@ func (o Options) validateVnetSubnetID() error {
 	return nil
 }
 
+func (o Options) validateNetworkDataplane() error {
+	if o.NetworkDataplane != "azure" && o.NetworkDataplane != "cilium" {
```
Comment (Bryce-Soghigian) [inline pkg/operator/options/options_validation.go:49]: these can move to using the constants package when that merges in

**Review by comtalyst — APPROVED** [review]
(discussed offline)

Approving code changes in general. 
But let's make sure we get things right on the relationship between `NetworkPolicy` and `NetworkDataplane`.

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

**Pull request merged**